packages/expo-web-browser/ios/WebBrowserModule.swift

```diff
@@ -7,23 +7,28 @@ import AuthenticationServices
 final public class WebBrowserModule: Module {
   private var currentWebBrowserSession: WebBrowserSession?
   private var currentAuthSession: WebAuthSession?
+  private var promise: Promise?
 
   public func definition() -> ModuleDefinition {
     Name("ExpoWebBrowser")
 
-    AsyncFunction("openBrowserAsync") { (url: URL, options: WebBrowserOptions, promise: Promise) throws in
-      guard self.currentWebBrowserSession?.isOpen != true else {
+    AsyncFunction("openBrowserAsync") { (url: URL, options: WebBrowserOptions, promise: Promise) in
+      guard self.promise == nil else {
         throw WebBrowserAlreadyOpenException()
       }
-      self.currentWebBrowserSession = WebBrowserSession(url: url, options: options) {
-        self.destroyBrowserSession()
+      self.currentWebBrowserSession = WebBrowserSession(url: url, options: options) { type in
+        self.promise?.resolve(["type": type])
+        self.promise = nil
+        self.currentWebBrowserSession = nil
       }
-      self.currentWebBrowserSession?.open(promise)
+      self.currentWebBrowserSession?.open()
+      self.promise = promise
```

Comment:
We probably don't have to keep the promise here. The `onDismiss` closure can just capture the local `promise` 🤔